CoreAardvark/ARKExceptionLogging.m

```diff
@@ -0,0 +1,53 @@
+//
+//  ARKExceptionLogging.m
+//  Aardvark
+//
+//  Created by Nick Entin on 9/7/17.
+//  Copyright 2017 Square, Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+@import CoreAardvark;
+
+#import "ARKExceptionLogging.h"
+
+
+NSUncaughtExceptionHandler *_Nullable ARKPreviousUncaughtExceptionHandler = nil;
+
+void ARKHandleUncaughtException(NSException *exception)
+{
+    ARKLogWithType(ARKLogTypeError, nil, @"Uncaught exception '%@':\n%@", exception.name, exception.debugDescription);
+    dispatch_semaphore_t logDistributionSemaphore = dispatch_semaphore_create(0);
+    [[ARKLogDistributor defaultDistributor] distributeAllPendingLogsWithCompletionHandler:^{
```

Comment:
From what I can tell, it won't reenter the exception handler if an exception occurs.

Added a call to `-[ARKLogDistributor distributeAllPendingLogsWithCompletionHandler:]` to bump the thread priority.  My goal here was to make sure that the final exception log gets written to disk (so only the default log distributor), not to flush all log distributors/observers.  Flushing all logs might be helpful though, since that could give context around the crash.

Reply:
Since adding that functionality to the log observers would be a breaking change, what do you think about making it only flush the default distributor for now, and file an issue to flush all observers too in 4.0?

Reply:
Filed #68 to track